tagging/utils.py

```diff
@@ -3,12 +3,25 @@
 calculation.
 """
 import math
-import types
 
 from django.db.models.query import QuerySet
-from django.utils.encoding import force_unicode
+try:
+    from django.utils.encoding import force_unicode
+except ImportError:
+    def force_unicode(s, *args, **kwargs):
+        return s
 from django.utils.translation import ugettext as _
 
+try:
+    basestring
+except NameError:  # Python 3 doesn't have basestring
+    basestring = str
+
+try:
+    long
+except NameError:  # Python 3 doesn't have long
+    long = int
```

Comment:
You may be able to clean this up by using one of the Py2/3 options (I think option1  doesn't need the future/past libs) here:  http://python-future.org/compatible_idioms.html#long-integers